Implement live data on delegate monitor - Closes #2563

slaweet opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### What issue have I solved?
<!--- Complementary description if needed -->
Closes #2563

### How have I implemented/fixed it?
<!--- Describe your technical implementation -->
- [x] Create `withForgingStatus` HOC and use it on delegates monitor
- [x] Fetch 101 latest blocks and add them to existing `blocks.latestBlocks` redux store
- [x] Insert `status` to each `delegate` in `withForgingStatus` HOC
- [x] Implement "Status" column
- [x] Implement "Status" column tooltips for active delegates
- [x] Fetch last blocks in "Status" column tooltips for "Not forging" delegates
- [x] Insert `forging time` to each `delegate` in `withForgingStatus` HOC
- [x] Implement "Forging time" column
- [x] Update "Forging time" with new blocks
- [x] Fix test coverage of `withForgingStatus`

### How has this been tested?
<!--- Please describe how you tested your changes. -->
- Go to https://jenkins.lisk.io/test/lisk-hub/PR-2569/#/monitor/delegates
- Interact with the page

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

osvaldovega commented on package.json:
```diff
@@ -214,10 +215,10 @@
     "webpack-bundle-analyzer": "3.0.3",
     "webpack-cli": "^3.3.8",
-    "webpack-dev-server": "^3.8.1"
+    "webpack-dev-server": "3.8.2"
```
hey Vit, I noticed that you update these packages version. what is perfect but why you did it? I mean did you check this with Yashar? is just for avoid any future problem.

slaweet replied:
These changes come from `npm update` because there was some warning telling me to do that. Is it ok @yasharAyari ? 

If not, I can just revert https://github.com/LiskHQ/lisk-hub/pull/2569/commits/212f06fa734c8ff667d4fc7e26217a827c893e4d

osvaldovega commented:
In the UI I found these differences and maybe you can check it.

🎨Is the size of the container ok when there is only one element
<img width="1446" alt="Screen Shot 2019-10-23 at 10 16 54" src="https://user-images.githubusercontent.com/11184770/67377396-e13f7a80-f585-11e9-813f-5264870a8a94.png">

🎨In testnet I got this error message I think it shouldn't be this the message.
<img width="1445" alt="Screen Shot 2019-10-23 at 10 08 52" src="https://user-images.githubusercontent.com/11184770/67377473-00d6a300-f586-11e9-925d-edc57a057457.png">

🎨In stand by delegates I got this error message, is this ok?
<img width="1389" alt="Screen Shot 2019-10-23 at 10 17 36" src="https://user-images.githubusercontent.com/11184770/67377537-206dcb80-f586-11e9-9425-f0a224d0ec5f.png">

🎨In the invision I see tooltip is showing up in the bottom but in the page is in the left
**this is inVision**
<img width="262" alt="Screen Shot 2019-10-23 at 10 21 58" src="https://user-images.githubusercontent.com/11184770/67377656-501cd380-f586-11e9-84c6-c06f1e0125d7.png">
**This is your branch**
<img width="533" alt="Screen Shot 2019-10-23 at 10 22 05" src="https://user-images.githubusercontent.com/11184770/67377673-56ab4b00-f586-11e9-99e5-5a381ca375e4.png">

🎨For the awaiting status, the tooltip show a different message, check it.
**inVision**
<img width="299" alt="Screen Shot 2019-10-23 at 10 23 26" src="https://user-images.githubusercontent.com/11184770/67377798-83f7f900-f586-11e9-8ffe-40162e20c4f4.png">
**Your Branch**
<img width="349" alt="Screen Shot 2019-10-23 at 10 23 16" src="https://user-images.githubusercontent.com/11184770/67377826-8e19f780-f586-11e9-8cc5-cccd845dab75.png">

🎨The last column the values should be bold.
inVision
<img width="117" alt="Screen Shot 2019-10-23 at 10 24 42" src="https://user-images.githubusercontent.com/11184770/67378776-1056eb80-f588-11e9-8e03-6b51729bca0d.png">

your branch
<img width="120" alt="Screen Shot 2019-10-23 at 10 24 48" src="https://user-images.githubusercontent.com/11184770/67378789-15b43600-f588-11e9-8fd5-817270fb278a.png">

🎨 The tooltip alignment in the hear are not align with the title
<img width="670" alt="Screen Shot 2019-10-23 at 10 26 44" src="https://user-images.githubusercontent.com/11184770/67378855-32506e00-f588-11e9-95fb-c070027a0dd3.png">

slaweet commented:
@imdominic 
Is the size of the container ok when there is only one element? We are using this on all pages that there is a minimum height of the box.
<img width="1446" alt="Screen Shot 2019-10-23 at 10 16 54" src="https://user-images.githubusercontent.com/11184770/67377396-e13f7a80-f585-11e9-813f-5264870a8a94.png">

osvaldovega commented:
@yasharAyari can you please take a look at this just to see if there is no problem about this.
So far is just update packages that we already have so shouldn't be a problem but we will like to ask you

osvaldovega commented:
Thanks Vit I saw the updates

slaweet commented:
@reyraa how should error messages work?

Is it ok to show the real error to the advanced users?
<img width="1445" alt="Screen Shot 2019-10-23 at 10 08 52" src="https://user-images.githubusercontent.com/11184770/67377473-00d6a300-f586-11e9-925d-edc57a057457.png">

reyraa commented:
@slaweet I would replace the URL with "Testnet/server". What do you think?

slaweet commented:
@osvaldovega 
- Implemented "Unable to connect ..." error according to @reyraa's suggestion. 
- Checked with @imdominic that the minimum box height on single row is as expected because otherwise, the box height is jumping too much.
- "Awaiting" status was working ok, but what you reported was supposed to be "Loading" status, so I changed the label from "Standby" to "Loading".

I addressed all other comments or yours.

osvaldovega commented:
Thanks Vit I checked and now all looks good

imdominic commented:
Thanks Vit from all the work, here are just some little visual issues that I and Julian discovered across the Delegate Monitor:

🎨 Tooltip in some cases stands on 3 lines and others 4 lines. If possible let's try to keep it on 3 lines as per design.

Screesnap below of the one with 4 lines of text:

![image (1)](https://user-images.githubusercontent.com/45934100/67473439-f7673c80-f652-11e9-9228-78e09bb1d821.png)

🎨 For the breakpoint, a design has been added to Invision, you can use this as a reference:

https://invis.io/ZRU30GY5NTX#/387849143_Monitor_-_Delegates_-_768_

We just contracted the address on the breakpoint iself, and I think would be ideal as per discussion with @slaweet.
